streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -93,6 +96,33 @@ long partitionTimestamp(final TopicPartition partition) {
         return queue.partitionTime();
     }
 
+    // creates queues for new partitions, removes old queues, saves cached records for previously assigned partitions
+    void updatePartitions(final Set<TopicPartition> newInputPartitions, final Function<TopicPartition, RecordQueue> recordQueueCreator) {
+        final Iterator<Map.Entry<TopicPartition, RecordQueue>> queuesIterator = partitionQueues.entrySet().iterator();
```

Comment:
nit: let's put iterator initialization closer to the start of while loop

Reply:
fixed